### PR TITLE
Add OCI hauler manifests.

### PR DESCRIPTION
**Pull request by amartin120**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Allows for carbide flavored hauler manifest to be utilized in an "easy button" format.

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
hauler store sync --product rancher:v2.78

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: cmd/hauler/cli/store/sync.go [hunk 1]
```diff
@@ -6,30 +6,37 @@ import (
...
 	ContentFiles []string
+	Key          string
+	Products	 []string
 }
 
 func (o *SyncOpts) AddFlags(cmd *cobra.Command) {
 	f := cmd.Flags()
 
 	f.StringSliceVarP(&o.ContentFiles, "files", "f", []string{}, "Path to content files")
+	f.StringVarP(&o.Key, "key", "k", "", "(Optional) Path to the key for image signature verification")
```
Comment (zackbradys) [inline cmd/hauler/cli/store/sync.go]:
```suggestion
	f.StringVarP(&o.Key, "key", "k", "", "(Optional) Path to the key for signature verification")
```

File: cmd/hauler/cli/store/sync.go [hunk 2]
```diff
@@ -6,30 +6,37 @@ import (
...
+	f.StringSliceVar(&o.Products, "products", []string{}, "Used for RGS Carbide customers to supply a product and version and Hauler will retrieve the images. i.e. '--product rancher=v2.7.8'")
```
Comment (zackbradys) [inline cmd/hauler/cli/store/sync.go]: rancher v2.7.8 not considered stable... might want to have latest stable of rancher v2.7.6 listed as the example
```suggestion
	f.StringSliceVar(&o.Products, "products", []string{}, "Used for RGS Carbide customers to supply a product and version and Hauler will retrieve the images. i.e. '--product rancher=v2.7.6'")
```

File: pkg/artifacts/file/getter/getter_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		},
+		{
+			name: "should identify a directory",
+			args: args{
+				source: rootDir,
+			},
+			want: "directory",
+		},
+		{
+			name: "should identify a http",
```
Comment (zackbradys) [inline pkg/artifacts/file/getter/getter_test.go]:
```suggestion
			name: "should identify an http fqdn",
```

File: pkg/artifacts/file/getter/getter_test.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+			args: args{
+				source: "http://my.cool.website",
+			},
+			want: "http",
+		},
+		{
+			name: "should identify a http",
```
Comment (zackbradys) [inline pkg/artifacts/file/getter/getter_test.go]:
```suggestion
			name: "should identify an http fqdn",
```

File: pkg/consts/consts.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+	// FileLayerMediaType is the reserved media type for File content layers
+	FileLayerMediaType = "application/vnd.content.hauler.file.layer.v1"
+
+	// FileLocalConfigMediaType is the reserved media type for File config
+	FileLocalConfigMediaType     = "application/vnd.content.hauler.file.local.config.v1+json"
+	FileDirectoryConfigMediaType = "application/vnd.content.hauler.file.directory.config.v1+json"
+	FileHttpConfigMediaType      = "application/vnd.content.hauler.file.http.config.v1+json"
+
+	// MemoryConfigMediaType
```
Comment (zackbradys) [inline pkg/consts/consts.go]: add description to "MemoryConfigMediaType is ..." to match others?

**Review by zackbradys — COMMENTED** [review]
few comments and suggested changes... feel free to take them or leave them!

**Review by dweomer — APPROVED** [review]

**Pull request merged**